common/lib/xmodule/xmodule/library_tools.py

```diff
@@ -27,39 +35,63 @@ def normalize_key_for_search(library_key):
 
 class LibraryToolsService:
     """
-    Service that allows LibraryContentBlock to interact with libraries in the
-    modulestore.
+    Service for LibraryContentBlock and LibrarySourcedBlock.
+
+    Allows to interact with libraries in the modulestore and blockstore.
     """
     def __init__(self, modulestore, user_id):
         self.store = modulestore
         self.user_id = user_id
 
-    def _get_library(self, library_key):
+    def _get_library(self, library_key, is_v2_lib):
         """
-        Given a library key like "library-v1:ProblemX+PR0B", return the
-        'library' XBlock with meta-information about the library.
+        Helper method to get either V1 or V2 library.
 
-        A specific version may be specified.
+        Given a library key like "library-v1:ProblemX+PR0B" (V1) or "lib:RG:rg-1" (v2), return the 'library'.
+
+        is_v2_lib (bool) indicates which library storage should be requested:
+            True - blockstore (V2 library);
+            False - modulestore (V1 library).
 
         Returns None on error.
         """
-        if not isinstance(library_key, LibraryLocator):
-            library_key = LibraryLocator.from_string(library_key)
-
-        try:
-            return self.store.get_library(
-                library_key, remove_version=False, remove_branch=False, head_validation=False
-            )
-        except ItemNotFoundError:
-            return None
+        if is_v2_lib:
+            try:
+                return library_api.get_library(library_key)
+            except ContentLibrary.DoesNotExist:
+                return None
+        else:
+            try:
+                return self.store.get_library(
+                    library_key, remove_version=False, remove_branch=False, head_validation=False
+                )
+            except ItemNotFoundError:
+                return None
```

Comment:
Here the same. I am just asking it is not obligatory

Reply:
See the comment above ^^